scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/TimedDoTaskAction.java

```diff
@@ -127,17 +122,9 @@ public Void call() throws Exception {
 
                 params = new TaskResult[parentIds.size()];
 
-                Map<TaskId, TaskResult> taskResults = new HashMap<>();
-                // Batch fetching of parent tasks results
-                for (List<TaskId> parentsSubList : ListUtils.partition(new ArrayList<>(parentIds),
-                                                                       PASchedulerProperties.SCHEDULER_DB_FETCH_TASK_RESULTS_BATCH_SIZE.getValueAsInt())) {
-                    taskResults.putAll(schedulingService.getInfrastructure()
-                                                        .getDBManager()
-                                                        .loadTasksResults(job.getId(), parentsSubList));
-                }
                 int i = 0;
                 for (TaskId taskId : parentIds) {
-                    params[i] = taskResults.get(taskId);
+                    params[i] = task.getParentTasksResults().get(taskId);
```

Comment:
Maybe, one thing that could be safer, would be to check the nullability of `task.getParentTasksResults()`. As the parentTaskResults are set much upstream, I am afraid that in some corner cases the parentTaskResults are not properly set (especially when the scheduler state is recovered). 

Then if `task.getParentTasksResults()` is null, we can fall back to the DB request. This way the DB request will not be executed in 99% of the cases, but we still have it if we need.

Reply:
After discussion with Amine, we agreed that it is preferable to check the nullability of `task.getParentTasksResults()` at this particular location.  We will add this control test.